cmds/webboot/webboot.go

```diff
@@ -207,6 +215,40 @@ func (d *DirOption) exec(uiEvents <-chan ui.Event) (menu.Entry, error) {
 	return menu.PromptMenuEntry("Distros", "Choose an option:", entries, uiEvents)
 }
 
+// If the chosen distro has a checksum, verify it
+// If the checksum is not correct, prompt the user to choose whether they still want to continue
+func displayChecksumPrompt(uiEvents <-chan ui.Event, supportedDistros map[string]Distro, label string, fpath string) (menu.Entry, error) {
+	// Check that the distro is supported
+	if _, ok := supportedDistros[label]; ok {
+		distro := supportedDistros[label]
+		// Check that checksum is available
+		if distro.checksum == "" {
+			accept, err := menu.PromptConfirmation("This distro does not have a checksum. Proceed anyway?", uiEvents)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to prompt confirmation: %s", err)
+			}
+			if !accept {
+				// Go back to download menu
+				return &DownloadOption{}, nil
+			}
+		} else if valid, calcChecksum, err := bootiso.VerifyChecksum(fpath, distro.checksum, distro.checksumType); err != nil {
+			return nil, fmt.Errorf("Failed to verify checksum: %s", err)
+		} else if !valid {
+			accept, err := menu.PromptConfirmation(fmt.Sprintf("Checksum was not correct. The correct checksum is %s and the downloaded ISO's checksum is %s. Proceed anyway?",
```

Comment:
This line is pretty long. Maybe split it up.

Reply:
Usually in Go, there is not line length limit besides keeping it reasonable. But, in the case, does the line get truncated or wrapped when it is printed in the UI? We don't want the user to see something cut off.